packages/roosterjs-editor-plugins/lib/HyperLink/HyperLink.ts

```diff
@@ -156,7 +161,7 @@ export default class HyperLink implements EditorPlugin {
             let target = this.target || '_blank';
             let window = this.editor.getDocument().defaultView;
             try {
-                window.open(href, target);
+                this.onLinkOpen(window, href, target, keyboardEvent);
```

Comment:
Instead of passing in window/href/target, I suggest pass in the A element so that you can get everything from this element and in case you need other attributes as well.

Reply:
good idea